src/privatesend-client.cpp

```diff
@@ -1326,14 +1332,16 @@ bool CPrivateSendClient::CreateDenominated(const CompactTallyItem& tallyItem, bo
         LogPrintf("CPrivateSendClient::CreateDenominated -- Error: %s\n", strFail);
         for(auto key : reservekeyDenomVec)
             key->ReturnKey();
-        reservekeyCollateral.ReturnKey();
+        if (fCreateMixingCollaterals)
+            reservekeyCollateral.ReturnKey();
         LogPrintf("CPrivateSendClient::CreateDenominated -- %d keys returned\n", reservekeyDenomVec.size() + 1);
```

Comment:
This log message is no longer accurate for `fCreateMixingCollaterals == false` (off by 1). The same is true for `KeepKey` part below too.